services-gateway-tests/src/main/java/io/scalecube/services/testservice/SecuredRsGwSessionHandler.java

```diff
@@ -0,0 +1,29 @@
+package io.scalecube.services.testservice;
+
+import io.scalecube.services.api.ServiceMessage;
+import io.scalecube.services.gateway.GatewaySession;
+import io.scalecube.services.gateway.SessionEventHandler;
+
+public class SecuredRsGwSessionHandler implements SessionEventHandler<ServiceMessage> {
```

Comment:
this calss loos very simular to ws gw session could it be just gateway session to serve both?

Reply:
not really just because the GatewayMessage vs ServiceMessage in mapMessage.

Reply:
@artem-v i think we can remove GatewayMessage looks like a duplicate

Reply:
I think ServiceMessage is more service.api... you can do nothing without ServiceMessage actually

Reply:
@snripa i know service message is required i was thinking that gateway message is not required.
you can construct the same structure with ServiceMessage.

Reply:
> but MAYBE better to do it in separate pull request.

ok. agree.